src/jit/lowerxarch.cpp

```diff
@@ -2690,6 +2712,22 @@ void Lowering::TreeNodeInfoInitSIMD(GenTree* tree)
     TreeNodeInfo* info     = &(tree->gtLsraInfo);
     LinearScan*   lsra     = m_lsra;
     info->dstCount         = 1;
+#ifdef FEATURE_AVX_SUPPORT
+    // check SIMD vector size, if it's 32 bytes: it's 256bit AVX instruction
+    // and set both flags on. If it's 16 bytes: set the ContainsAVX flag
+    if (comp->getSIMDInstructionSet() == InstructionSet_AVX)
+    {
+        if (simdTree->gtSIMDSize == 32)
+        {
+            comp->codeGen->getEmitter()->SetContainsAVX(true);
+            comp->codeGen->getEmitter()->SetContains256bitAVX(true);
+        }
+        else if (simdTree->gtSIMDSize == 16)
+        {
+            comp->getEmitter()->SetContainsAVX(true);
```

Comment:
Suggestion: this could be factored out as it is always done. Also, if we have any SIMD node, then we will be using AVX-encoded instructions. Note that the gtSIMDSize could be 8 or 12 for a fixed-size vector type. I think that you should unconditionally call `SetContainsAVX(true)` and then only call `SetContains256bitAVX(true)` for the 32 byte size.